src/tools/rbd/action/Device.cc

```diff
@@ -189,6 +205,43 @@ int execute_unmap(const po::variables_map &vm,
   return (*get_device_operations(vm)->execute_unmap)(vm, ceph_global_init_args);
 }
 
+void get_attach_arguments(po::options_description *positional,
+                          po::options_description *options) {
+  add_device_type_option(options);
+  at::add_image_or_snap_spec_options(positional, options,
+                                     at::ARGUMENT_MODIFIER_NONE);
+  options->add_options()
+    ("device", po::value<std::string>(), "specify nbd device path")
+    ("read-only", po::bool_switch(), "map read-only")
+    ("exclusive", po::bool_switch(), "disable automatic exclusive lock transitions")
+    ("quiesce", po::bool_switch(), "use quiesce hooks")
+    ("quiesce-hook", po::value<std::string>(), "quiesce hook path");
```

Comment:
Just a note, there are other options with rbd-nbd which we can consider to add here, like:

```
  --encryption-format           Image encryption format
                                (possible values: luks1, luks2)
  --encryption-passphrase-file  Path of file containing passphrase for unlocking image encryption
  --io-timeout <sec>            Set nbd IO timeout
  --max_part <limit>            Override for module param max_part
  --nbds_max <limit>            Override for module param nbds_max
  --reattach-timeout <sec>      Set nbd re-attach timeout
                                (default: 30)
  --try-netlink                 Use the nbd netlink interface
```

Reply:
Yes, I very much understand that part @trociny
just giving a hint to reviewers here as couldn't make the decision whether to add them or not.

Thanks!

Reply:
> The rbd-nbd specific options can be provided via `--options`. See `add_device_specific_options` just below.

Agree with this.